src/movies.js

```diff
@@ -1,16 +1,163 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(movies) {
+    let newArray = movies.map(function(movie){
+        return movie.director; 
+    })
+    return newArray;
+}
+
+
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+function cleanArray (newArray) {
+    uniqueDirectors = [];
+    for (let el in newArray) {
+        if (uniqueDirectors.indexOf(el) < 0) uniqueDirectors.push(el);
+    }
+    return uniqueDirectors;
+}
+
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies (movies) {
+    
+    const totalMovies = movies.filter(function (item) {
+        const isSteven = item.director === 'Steven Spielberg';
+        const isDrama = item.genre.indexOf('Drama') >= 0;
+        return isSteven && isDrama;
+    });
+    return totalMovies.length;
+}
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage (movies){
+    if (movies.length === 0) return 0;
+    const ratedMovies = movies.filter(function(item){
+        return item.rate > 0;
+    })
+    const sum = ratedMovies.reduce(function (total, movie){
+        return total + movie.rate;
+    }, 0)
+    const avg = sum / movies.length;
+    return +avg.toFixed(2);
+}
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(movies) {
+    const dramaMovies = movies.filter(function(item){
+        const isDrama = item.genre.indexOf('Drama') >= 0;
+        return isDrama;
+    })
+    return ratesAverage (dramaMovies);
+    
+}
+
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
 
+function orderByYear(movies) {
+    const newArray = movies.map(function(item){
+        return item;
+    })
+    const oldestFirst = newArray.sort(function(a, b){
+        if (a.year === b.year) {
+            if (a.title > b.title) {
+                return 1;
+            } 
+            if (a.title < b.title) {
+                return -1;
+            }
+            return 0;
+        }
+        return a.year - b.year;  
+    }); 
+    return oldestFirst;
+}
+
```

Comment:
again here, you don't need to add the extra functionality in the original method to get the same answer, for example: 

```
function sortIt(arr) {
  return arr.sort((a,b) => 
    a - b
  )
}
         
sortIt([1,5,7,9,6,2,5,7,])
```